api/v1beta1/provider_types.go

```diff
@@ -48,11 +48,14 @@ type ProviderCredentials struct {
 
 // ProviderStatus defines the observed state of Provider.
 type ProviderStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	State   string `json:"state,omitempty"`
```

Comment:
Suggest:

```go
type ProviderPhase string

const (
	// ProviderPhaseReady is the `ready` state
	ProviderPhaseReady ProviderPhase = "ready"
	// ProviderPhaseInitializing marks the state of a Provider is initializing
	ProviderPhaseInitializing ProviderPhase = "initializing"
)

type ProviderStatus struct {
	State   ProviderPhase `json:"state,omitempty"
        Message string `json:"message,omitempty"`
}
```

Reply:
Fixed in https://github.com/oam-dev/terraform-controller/pull/98/commits/68d38c09bf1370aa7c75e900d12163ca0b311bc5. Thanks.